src/rust/lib/flexer/src/group.rs

```diff
@@ -0,0 +1,256 @@
+use crate::automata::pattern::Pattern;
+use crate::automata::nfa::NFA;
+use crate::group::rule::Rule;
+
+use itertools::Itertools;
+
+pub mod rule;
+
+/// Group of a set of rules.
+struct Group {
+    /// ID of group.
+    pub id     : usize,
+    /// Name of group.
+    pub name   : String,
+    /// Parent of group.
+    pub parent : Option<Box<Group>>,
+    /// Set of rules. See Rule for more information.
+    pub rules  : Vec<Rule>,
+    /// Function that is called on exiting group.
+    pub finish : Box<dyn FnMut()>,
+}
+
+impl Default for Group {
+    fn default() -> Self {
+        Group {
+            name   : Default::default(),
+            id     : Default::default(),
+            parent : Default::default(),
+            rules  : Default::default(),
+            finish : Box::new(||{}),
+        }
+    }
+}
+
+impl Group {
+
+    /// Adds new rule to group.
+    pub fn add_rule(&mut self, rule:Rule) {
+        self.rules.push(rule)
+    }
```

Comment:
use `Add` trait

Reply:
That doesn't make sense. `add_rule` modifies group.